### PR TITLE
feat: implement zkevm ecosystem

### PR DESCRIPTION
**Pull request by bilbeyt**
### What I did

fixes: #

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: tests/test_provider.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+def test_basic(accounts, networks):
+    with networks["polygon-zkevm"].local.use_provider("test"):
```
[the remark this answers is not in the document]
Comment (antazoey) [reply tests/test_provider.py:2]: potentially put in a fixture too

**Review by antazoey — COMMENTED** [review]
tests failing because `typing-extensions` broke pydantic 15 hours ago and everything fails with this error now.
for a temp fix, pin `typing-extensions==4.5.0`
and maybe put a comment saying you can remove it once pydantic patched and ape updated its min pydantic version

**Comment by bilbeyt** [comment]
@antazoey pr seems ready,  will check again ?

File: ape_polygon_zkevm/ecosystem.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            )
+
+        return txn_class.parse_obj(kwargs)
+
+
+def _get_transaction_type(_type: Optional[Union[int, str, bytes]]) -> TransactionType:
+    if not _type:
+        return TransactionType.STATIC
+
+    if _type is None:
```
Comment (antazoey) [inline ape_polygon_zkevm/ecosystem.py:98]: Is it intentional to use STATIC fee as the default? Is this chain non-EIP-1559? Just want to make sure not a copy-paste error

File: tests/test_provider.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+def test_basic(accounts, local_network):
+    with local_network.use_provider("test"):
```
Comment (antazoey) [inline tests/test_provider.py:2]: Could move the connection to a contexted-fixture

File: setup.py [hunk 4]
```diff
@@ -46,25 +48,27 @@
...
     python_requires=">=3.8,<4",
     extras_require=extras_require,
-    py_modules=["<MODULE_NAME>"],
+    py_modules=["ape_polygon_zkevm"],
     license="Apache-2.0",
     zip_safe=False,
     keywords="ethereum",
     packages=find_packages(exclude=["tests", "tests.*"]),
-    package_data={"<MODULE_NAME>": ["py.typed"]},
+    package_data={"ape_polygon_zkevm": ["py.typed"]},
```
Comment (antazoey) [inline setup.py:71]: Could add python 3.11 flair as it should also work on that

File: ape_polygon_zkevm/ecosystem.py (hunk 2 above)
Comment (antazoey) [reply ape_polygon_zkevm/ecosystem.py:98]: thanks for checking!

**Comment by antazoey** [comment]
i think the readme still has template stuff in it? may want to do a CTRL-F

**Comment by bilbeyt** [comment]
> i think the readme still has template stuff in it? may want to do a CTRL-F

README is also done

**Comment by bilbeyt** [comment]
can i merge this now @antazoey ? then transfer it

File: tests/conftest.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+def runner():
+    return CliRunner()
+
+
+@pytest.fixture
+def cli():
+    return ape_cli
+
+
+@pytest.fixture()
```
Comment (antazoey) [inline tests/conftest.py:27]:
```suggestion
@pytest.fixture
```

**Review by antazoey — APPROVED** [review]

**Comment by antazoey** [comment]
the typing ext pin is no longer needed i dont think, now that pydantic has been patched.

**Comment by bilbeyt** [comment]
> the typing ext pin is no longer needed i dont think, now that pydantic has been patched.

Pin removed on all plugins @antazoey ape-base ape-zksync and this.

**Pull request merged**